src/main.rs

```diff
@@ -1311,78 +1311,95 @@ fn parse_manpage_at_path(
-//
-//         # Now skip if requested
-//         if skip:
-//             continue
-//
-//         try:
-//             if parse_manpage_at_path(manpage_path, output_directory):
-//                 successful_count += 1
-//         except IOError:
-//             diagnostic_indent = 0
-//             add_diagnostic('Cannot open ' + manpage_path)
-//         except (KeyboardInterrupt, SystemExit):
-//             raise
-//         except:
-//             add_diagnostic('Error parsing %s: %s' % (manpage_path, sys.exc_info()[0]), BRIEF_VERBOSE)
-//             flush_diagnostics(sys.stderr)
-//             traceback.print_exc(file=sys.stderr)
-//         flush_diagnostics(sys.stderr)
-//     print("") #Newline after loop
-//     add_diagnostic("Successfully parsed %d / %d pages" % (successful_count, total_count), BRIEF_VERBOSE)
-//     flush_diagnostics(sys.stderr)
-
-#[derive(Copy, Clone, Debug)]
-struct Progress(pub bool);
-
-// TODO Arg/output types?
+/// Get the number of digits in num
+fn num_digits(n: usize) -> usize {
+    (1.max(n) as f32).log10() as usize + 1
+}
+
+#[test]
+fn test_num_digits() {
+    assert_eq!(num_digits(1000), 4);
+    assert_eq!(num_digits(100), 3);
+    assert_eq!(num_digits(33), 2);
+    assert_eq!(num_digits(123456789012345), 15);
+    assert_eq!(num_digits(0), 1);
+}
+
 fn parse_and_output_man_pages(
-    _paths: impl Iterator<Item = PathBuf>,
-    _output_directory: PathBuf,
-    Progress(_show_progress): Progress,
+    paths: &mut [PathBuf],
+    output_directory: PathBuf,
+    show_progress: bool,
+    deroff_only: bool,
+    write_to_stdout: bool,
 ) {
-    unimplemented!();
+    paths.sort();
+
+    let total = paths.len();
+
+    let mut successful_count = 0;
+    let padding = num_digits(total);
+
+    let mut last_len: usize = 0;
+
+    if show_progress && !write_to_stdout {
+        println!(
+            "Parsing man pages and writing completions to {:?}",
+            output_directory
+        );
+    }
+
+    for (index, manpage_path) in paths.iter().enumerate() {
+        // foo/bar/gcc.1.gz -> gcc.1.gz
+        let man_file_name = manpage_path
+            .file_name()
+            .map(|fname| fname.to_string_lossy())
+            .unwrap_or_else(|| panic!("Failed to get manfile name from {:?}", manpage_path));
+
+        // gcc.1.gz -> gcc
+        // `str::split` iterator ALWAYS has a first element by definition
+        let cmd_name = man_file_name.split('.').next().unwrap();
+
+        if show_progress && !write_to_stdout {
+            let progress = format!(
+                "  {0:>1$} / {2} : {3}",
+                index + 1,
+                padding,
+                total,
+                man_file_name,
+            );
+
+            let padded = format!("{0:<1$}", progress, last_len);
+
+            last_len = progress.len();
+            let stdout = std::io::stdout();
+            let mut lock = stdout.lock();
+            lock.write_all(format!("\r\x1b[K{}", padded).as_bytes())
```

Comment:
Are you sure we need to pad? Why not just use `progress` here?

Reply:
AFAIK the original did it to overwrite the last line with whitespace before printing again

Reply:
This already does that
```rust
use std::io::{self, Write};
use std::thread;
use std::time::Duration;

fn main() {
    for i in (0..103).rev() {
        let stdout = io::stdout();
        let mut stdout = stdout.lock();
        stdout.write_all(format!("\r\x1b[K{}", if i == 100 { 9999999 } else { i }).as_bytes()).ok();
        stdout.flush().ok();
        thread::sleep(Duration::from_secs(1));
    }
    println!();
}
```

Reply:
Ah, ok, I'll add that tonight

Reply:
69bca77